contracts/VestingEscrow.vy

```diff
@@ -0,0 +1,369 @@
+    if time >= self.disabled_at:
+        return 0
+    return self.total_locked - self._total_vested_at(time)
+
+
+@external
+@view
+def locked() -> uint256:
+    """
+    @notice Get the number of locked tokens for recipient
+    """
+    return self._locked()
+
+
+@external
+def claim(
+    beneficiary: address = msg.sender, amount: uint256 = max_value(uint256)
+):
+    """
+    @notice Claim tokens which have vested
+    @param beneficiary Address to transfer claimed tokens to
+    @param amount Amount of tokens to claim
+    """
+    self._check_sender_is_recipient()
+
+    claimable: uint256 = min(self._unclaimed(), amount)
+    self.total_claimed += claimable
+
+    assert self.token.transfer(
+        beneficiary, claimable, default_return_value=True
+    ), "transfer failed"
+
+    log Claim(self.recipient, beneficiary, claimable)
+
+
+@external
+def revoke_unvested():
+    """
+    @notice Disable further flow of tokens and revoke the unvested part to owner
+    """
+    self._check_sender_is_owner_or_manager()
+
+    # NOTE: Revoking more than once is futile
+    revokable: uint256 = self._locked()
+    self.disabled_at = block.timestamp
+
+    assert self.token.transfer(
+        self._owner(), revokable, default_return_value=True
+    ), "transfer failed"
+
+    log UnvestedTokensRevoked(msg.sender, revokable)
+
+
+@external
+def revoke_all():
+    """
+    @notice Disable further flow of tokens and revoke all tokens to owner
+    """
+    self._check_sender_is_owner()
+    assert self.is_fully_revokable, "not allowed for ordinary vesting"
+
+    # NOTE: Revoking more than once is futile
+    self.is_fully_revoked = True
+    self.disabled_at = block.timestamp
+    # NOTE: do not revoke extra tokens
+    revokable: uint256 = self.total_locked - self.total_claimed
+
+    assert self.token.transfer(
+        self._owner(), revokable, default_return_value=True
+    ), "transfer failed"
+
+    log VestingFullyRevoked(msg.sender, revokable)
+
+
+@external
+def recover_erc20(token: address, amount: uint256):
+    """
+    @notice Recover ERC20 tokens to recipient
+    @param token Address of the ERC20 token to be recovered
+    @param amount Amount of the ERC20 token to be recovered
+    """
+    recoverable: uint256 = amount
+    if token == self.token.address:
+        available: uint256 = ERC20(token).balanceOf(self) - (
+            self._locked() + self._unclaimed()
+        )
+        recoverable = min(recoverable, available)
+    if recoverable > 0:
+        assert ERC20(token).transfer(
+            self.recipient, recoverable, default_return_value=True
+        ), "transfer failed"
+        log ERC20Recovered(token, recoverable)
+
+
+@external
+def recover_ether():
+    """
+    @notice Recover Ether to recipient
+    """
+    amount: uint256 = self.balance
+    self._safe_send_ether(self.recipient, amount)
+    log ETHRecovered(amount)
+
+
+@external
+def aragon_vote(abi_encoded_params: Bytes[1000]):
```

Comment:
why `Bytes[1000]`?

Reply:
Just to leave margin for extra args in the new implementations
RN we have 2 args in the https://github.com/lidofinance/lido-vesting-escrow/blob/c72fa6eca01863cc7451e701a14df8a697b77d8b/contracts/VotingAdapter.vy#L75